db_stress_tool/expected_state.cc

```diff
@@ -303,7 +304,8 @@ Status FileExpectedStateManager::SaveAtAndAfter(DB* db) {
     trace_opts.filter |= kTraceFilterIteratorSeek;
     trace_opts.filter |= kTraceFilterIteratorSeekForPrev;
     trace_opts.preserve_write_order = true;
-    s = db->StartTrace(trace_opts, std::move(trace_writer));
+    s = db->StartTrace(trace_opts, std::move(trace_writer),
+                       initial_tracked_contents);
```

Comment:
Wished that i don't have to change StartTrace API but the underlying DB::tracer_ is write a formatted payload instead of just a slice so I believe we have to be consistent with that. I tried `trace_writer->Write(initial_tracked_contents)` before `db->StartTrace` , resulting in `corrupted trace` where we can't read the header (possibly taken up by our direct `trace_writer->Write(initial_tracked_contents)` at the beginning.)